pkg/graph/schema/conversion_cel_type_test.go

```diff
@@ -0,0 +1,290 @@
+// Copyright 2025 The Kubernetes Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package schema
+
+import (
+	"testing"
+
+	"github.com/google/cel-go/cel"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	extv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	apiservercel "k8s.io/apiserver/pkg/cel"
+	"k8s.io/apiserver/pkg/cel/openapi"
+	"k8s.io/kube-openapi/pkg/validation/spec"
+	"k8s.io/utils/ptr"
+)
+
+func TestInferSchemaFromCELType_Primitives(t *testing.T) {
+	tests := []struct {
+		name    string
+		celType *cel.Type
+		want    *extv1.JSONSchemaProps
+	}{
+		{
+			name:    "convert bool",
+			celType: cel.BoolType,
+			want:    &extv1.JSONSchemaProps{Type: "boolean"},
+		},
+		{
+			name:    "convert int",
+			celType: cel.IntType,
+			want:    &extv1.JSONSchemaProps{Type: "integer"},
+		},
+		{
+			name:    "convert uint",
+			celType: cel.UintType,
+			want:    &extv1.JSONSchemaProps{Type: "integer"},
+		},
+		{
+			name:    "convert double",
+			celType: cel.DoubleType,
+			want:    &extv1.JSONSchemaProps{Type: "number"},
```

Comment:
Curious if you've run into trouble with floating point precision causing flip flops in graph evaluation.

Reply:
Haven't seen it in practice yet, but yeah, could definitely cause issues. Any float comparisons expressions in resource templates, readyWhen and includeWhen could cause issues... worth documenting this to steering users towards using more of string/bool/int expression rather than floats, but it's inevitable i guess.